### ISSUE TITLE
Update and extend Cromer-Mann parameters for calculating scattering intensities

### ISSUE THREAD

**Issue by hejamu** [maicos-devel/maicos#105]
In GitLab by @PicoCentauri on Mar 13, 2024, 07:25 UTC

For computing the form factor we use an heuristic fit done by Cromer and Mann. They conducted Hartree-Fock simulations and obtained these parameters that we store in `src/maicos/share/sfactor.dat`. The number of digits is not very large and we should extend this similar to what the GROMACS project has in [their current version](https://gitlab.com/gromacs/gromacs/-/blob/main/share/top/scatteringfactors.dat?ref_type=heads).

Also, we should add the parameters for charged elements (ions) which are missing as well in the GROMACS code. If we add the ions we have to update the type-to-element mapping in `src/maicos/share/atomtypes.dat`, because currently we map ion types to their neutral elements, which is questionable. All values are reported [online](https://onlinelibrary.wiley.com/iucr/itc/Cb/ch6o1v0001/contents/) as PDF and HTML. See Table 6.1.1.4.

Basically, I have these tasks in mind

1. Rename `sfactor.dat` -> `scatteringfactors.dat` to have a better name
2. Update values according to GROMACS version
3. Add values for ions
4. Update reference in header accordingly

*Migrated from GitLab: https://gitlab.com/maicos-devel/maicos/-/issues/127*
